### PR TITLE
Use applications db/migrate paths when available

### LINKED ISSUE #22261: ActiveRecord::Migrator hardcodes 'db/migrate' path
https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/migration.rb#L874

``` ruby
def migrations_paths
  @migrations_paths ||= ['db/migrate']
  # just to not break things if someone uses: migration_path = some_string
  Array(@migrations_paths)
end
```

This results in `ActiveRecord::Migration.maintain_test_schema!` constantly triggering `rake db:test:prepare`, because it can't find any migration files under `db/migrate`

### PR DESCRIPTION
**Pull request by natesalisbury**
`ActiveRecord::Migrator::migrations_paths` defaults to 'db/migrate' when no other `migrations_paths` have been set. If your application uses a folder other than db/migrate for migrations, `ActiveRecord::Migration.maintain_test_schema!` in rails/test_help will trigger `rake db:test:prepare` at the start of each test run. 

When running database rake tasks this usually isn't an issue because they are dependent on the db:load_config task which will set `migrations_paths` based on an application's config. This is an issue however when calling `ActiveRecord::Migration.maintain_test_schema!` from rails/test_help which doesn't load the app's config, which then triggers db:test:prepare since `any_migrations?` will return false.

I noticed this after upgrading my application to rails 4.2 and each time I ran my cucumber tests, `rake db:test:prepare` was getting triggered at the start. This application has a large db schema so it was adding about 30 seconds to the beginning of each run.  

Fixes #22261

### CODE REVIEW

**Comment by maclover7** [comment]
@natesalisbury Can you add a regression test, to confirm to new behavior?

**Comment by natesalisbury** [comment]
@maclover7 done. Let me know if you need anything else.

File: activerecord/CHANGELOG.md
Comment (maclover7) [inline activerecord/CHANGELOG.md]: Can you add backticks around db/migrate (`db/migrate`)

File: activerecord/CHANGELOG.md
Comment (maclover7) [inline activerecord/CHANGELOG.md]: Same comment as above

File: activerecord/CHANGELOG.md
Comment (maclover7) [inline activerecord/CHANGELOG.md]: Can you add a period after the number

**Review by maclover7 — CHANGES_REQUESTED** [review]
Small CHANGELOG.md edits, otherwise looks good to me.

**Comment by natesalisbury** [comment]
Updated CHANGELOG.md.

**Review by maclover7 — APPROVED** [review]
👍 Awesome work!

**Comment by einarj** [comment]
Awesome @natesalisbury , I can't wait until this is merged.

**Comment by matthewd** [comment]
This has to be set in the Railtie instead of pulling it directly here... it's a layering violation for AR-the-library to go find the application and ask it for a value.

Can we perhaps move it from the rake task into the railtie proper? It seems rather odd to me that `DatabaseTasks` is the authoritative source for this, and we then copy from there to Migrator. :confused:

**Comment by eileencodes** [comment]
Closing due to inactivity and I believe the problem this is trying to fix is fixed by multiple database migration support in Rails 6.0.

**Pull request closed without merging**
